### PR TITLE
[TE] Set the size of heatmap using other metric

### PR DESCRIPTION
**Pull request by jihaozh**
This PR allows the user to select a different metric than the primary metric to set the cell size of RCA heatmap.

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js [hunk 1]
```diff
@@ -41,10 +47,16 @@ export default Component.extend({
 
   breakdowns: null, // {}
 
-  selectedUrn: null, // Set
+  selectedUrn: null, // ""
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js]: Can we remove all these comments? Do we need them?

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js [hunk 2]
```diff
@@ -56,6 +68,29 @@ export default Component.extend({
    */
   selectedMode: Object.keys(ROOTCAUSE_MODE_MAPPING)[1],
 
+  /**
+   * Selected option for heatmap size
+   */
+  sizeOptionSelected: computed('sizeMetricUrn', 'sizeOptions', {
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js]: can you not use this style for computed? Especially if you're not doing anything with set(). Try this
```
  sizeOptionSelected: computed(
    'sizeMetricUrn', 'sizeOptions',
    function() {
      const { sizeMetricUrn, sizeOptions } =
        this.getProperties('sizeMetricUrn', 'sizeOptions');
      return sizeOptions.find(opt => opt.urn === sizeMetricUrn);
    }
  ),
```

Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js]: try

```
 sizeOptionSelected: computed(
    'sizeMetricUrn', 'sizeOptions',
    function() {
      const { sizeMetricUrn, sizeOptions } =
        this.getProperties('sizeMetricUrn', 'sizeOptions');
      return sizeOptions.find(opt => opt.urn === sizeMetricUrn);
    }
  ).readOnly(),
```

Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js]: https://m.alphasights.com/the-risks-you-take-when-you-use-ember-computed-oneway-f2c008a14404

Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js]: this works. thanks for the example!

Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js]: Ideally we should use `oneWay` as readOnly will throw an error if someone (like a component) tries to mutate the object. OneWay makes a copy of the computed object and pass it on.

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js (hunk 1 above)
Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js]: removed

**Pull request merged**